sdk/component/enricher.go

```diff
@@ -0,0 +1,51 @@
+package component
+
+import (
+	"context"
+	"fmt"
+)
+
+// RunEnricher runs an enricher after initialising the run context.
+func RunEnricher(ctx context.Context, enricher Enricher, opts ...RunnerOption) error {
+	return run(
+		ctx,
+		func(ctx context.Context) error {
+			logger := LoggerFromContext(ctx).With(logKeyComponentType, "enricher")
+
+			logger.Debug("preparing to execute enricher component...")
```

Comment:
What do you think about adding the component name to the log? Might be useful if someone is debugging multiple components and trying to find out which one fails.

Reply:
This is done by default for every runner [here](https://github.com/smithy-security/smithy/blob/5d6f911170ba22fbb16a6cea58f3e759ad61c6b5/sdk/component/runner.go#L73) :)

That's the power of centralising into a single runner!